### PR TITLE
LPS-147923 [CP 2.0] Logic to skip DXP Cloud Activation on onboarding flow

### PR DESCRIPTION
**Pull request by igordmf**

### CODE REVIEW

**Review by Lucas2691 — APPROVED** [review]
Nice job, man!

**Review by iagochaves — APPROVED** [review]

**Review by thaynam — APPROVED** [review]

**Review by joaosvassis — APPROVED** [review]
Very good!

**Review by joanagrego — APPROVED** [review]
Nice job, Igor! Just one thing, i missed the PR description.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/reducer.js [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 export const actionTypes = {
 	CHANGE_STEP: 'CHANGE_STEP',
+	UPDATE_DXPCLOUD_ACTIVATION_STATUS: 'UPDATE_DXPCLOUD_ACTIVATION_STATUS',
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/reducer.js:14]: Action name should be an action together with state name
```suggestion
	UPDATE_DXP_CLOUD_ACTIVATION_SUBMITTED_STATUS: 'UPDATE_DXP_CLOUD_ACTIVATION_SUBMITTED_STATUS',
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/reducer.js [hunk 2]
```diff
@@ -49,6 +50,12 @@ const reducer = (state, action) => {
 				subscriptionGroups: action.payload,
 			};
 		}
+		case actionTypes.UPDATE_DXPCLOUD_ACTIVATION_STATUS: {
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/reducer.js:53]:
```suggestion
		case actionTypes.UPDATE_DXP_CLOUD_ACTIVATION_SUBMITTED_STATUS: {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/index.js [hunk 3]
```diff
@@ -123,6 +125,25 @@ const AppContextProvider = ({assetsPath, children}) => {
...
+					scopeKey: Liferay.ThemeDisplay.getScopeGroupId(),
+				},
+			});
+
+			if (data) {
+				const status = !!data.c?.dXPCloudEnvironments?.items.length;
+
+				dispatch({
+					payload: status,
+					type: actionTypes.UPDATE_DXPCLOUD_ACTIVATION_STATUS,
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/index.js:142]:
```suggestion
					type: actionTypes.UPDATE_DXP_CLOUD_ACTIVATION_SUBMITTED_STATUS,
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/index.js [hunk 4]
```diff
@@ -123,6 +125,25 @@ const AppContextProvider = ({assetsPath, children}) => {
...
+			const {data} = await client.query({
+				query: getDXPCloudEnvironment,
+				variables: {
+					filter: `accountKey eq '${accountKey}'`,
+					scopeKey: Liferay.ThemeDisplay.getScopeGroupId(),
+				},
+			});
+
+			if (data) {
+				const status = !!data.c?.dXPCloudEnvironments?.items.length;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/index.js:138]:
```suggestion
				const status = !!data.c?.dXPCloudEnvironments?.items?.length;
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/reducer.js [hunk 5]
```diff
@@ -49,6 +50,12 @@ const reducer = (state, action) => {
...
+			return {
+				...state,
+				DXPCloudActivationSubmittedStatus: action.payload,
```
[the remark this answers is not in the document]
Comment (igordmf) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/reducer.js]: Done! Thank you!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/reducer.js (hunk 2 above)
Comment (igordmf) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/context/reducer.js]: Done! Thank you!

**Comment by jairmedeiros** [comment]
ci:forward

**Review by jairmedeiros — APPROVED** [review]

**Pull request closed without merging**